scipy/stats/biasedurn.pyx.templ

```diff
@@ -54,42 +58,68 @@ cdef class _PyWalleniusNCHypergeometric:
         self.c_fnch.get().moments(&mean, &var)
         return mean, var
 
-
-cdef bitgen_t* make_rng(random_state=None):
-    # get a bit_generator object
-    if random_state is None or isinstance(random_state, int):
-        bg = np.random.RandomState(random_state)._bit_generator
-    elif isinstance(random_state, np.random.RandomState):
-        bg = random_state._bit_generator
-    elif isinstance(random_state, np.random.Generator):
-        bg = random_state.bit_generator
-    else:
-        raise ValueError('random_state is not in {None, int, RandomState, Generator}')
-
-    # get the bitgen_t pointer
-    cdef const char *capsule_name = "BitGenerator"
-    capsule = bg.capsule
-    if not PyCapsule_IsValid(capsule, capsule_name):
-        raise ValueError("Invalid pointer to anon_func_state")
-    cdef bitgen_t* crng = <bitgen_t *> PyCapsule_GetPointer(capsule, capsule_name)
-    return crng
+IF NPY_OLD:
+    cdef object _glob_rng
+    cdef double next_double(void* state) nogil:
+        with gil:
+            return _glob_rng.uniform()
+ELSE:
+    cdef bitgen_t* make_rng(random_state=None):
+        # get a bit_generator object
+        if random_state is None or isinstance(random_state, int):
+            bg = np.random.RandomState(random_state)._bit_generator
+        elif isinstance(random_state, np.random.RandomState):
+            bg = random_state._bit_generator
+        elif isinstance(random_state, np.random.Generator):
+            bg = random_state.bit_generator
+        else:
+            raise ValueError('random_state is not one of None, int, RandomState, Generator')
+        capsule = bg.capsule
+
+        # get the bitgen_t pointer
+        cdef const char *capsule_name = "BitGenerator"
+        if not PyCapsule_IsValid(capsule, capsule_name):
+            raise ValueError("Invalid pointer to anon_func_state")
+        cdef bitgen_t* crng = <bitgen_t *> PyCapsule_GetPointer(capsule, capsule_name)
+        return crng
 
 
 cdef class _PyStochasticLib3:
     cdef unique_ptr[StochasticLib3] c_sl3
+    IF not NPY_OLD:
+        cdef bitgen_t* bit_generator
 
     def __cinit__(self):
         self.c_sl3 = unique_ptr[StochasticLib3](new StochasticLib3(0))
+        IF NPY_OLD:
+            self.c_sl3.get().next_double = &next_double
+        ELSE:
+            self.c_sl3.get().next_double = self.bit_generator.next_double
```

Comment:
Might be complaining about this -- `bit_generator` is not initialized here.  I assumed that the compiler would be smart enough not to try to access the location, just point to it